internal/record/log_writer_test.go

```diff
@@ -66,6 +66,27 @@ func TestSyncQueue(t *testing.T) {
 	flusherWG.Wait()
 }
 
+func TestSyncErrorPropagation(t *testing.T) {
```

Comment:
Let me also assert the behaviour that sync requests receive the err even after flushLoop closes.